### PR TITLE
Repl support

### PR DESCRIPTION
**Pull request by Ghibranalj**
Issue #53 

I implemented it with https://github.com/chzyer/readline.

What i did :
- I refactored line 469 to 503 into a function called `runQuery()`
- I added a new field in args called `isInteractive`
- I added a new function `repl()`

If `args.isInteractive` is true `repl()` is called otherwise `runQuery()` is run directly

```go
if args.isInteractive {
	return repl(project, &ec, args, files)
}
return runQuery(lastNonFlagArg, project, &ec, args, files)
```

### CODE REVIEW

**Comment by Ghibranalj** [comment]
`name 'printf' is not defined` on macos thats weird.

File: main.go [hunk 1]
```diff
@@ -219,6 +220,98 @@ func importFile(projectId string, file, mimetype string, ec runner.EvalContext)
...
+
+	defer l.Close()
+
+	for {
+
+		queryRaw, err := l.Readline()
+		if err != nil {
+			return err
+		}
+		if queryRaw == "exit\n" || queryRaw == "EOF\n" {
```
Comment (eatonphil) [inline main.go:303]: These might not work on Windows since Windows produces \r\n not \n

Comment (Ghibranalj) [reply main.go:303]: oh yeah. it can be removed completely since its handled by readline already

Comment (Ghibranalj) [reply main.go:303]: oh yeah. it can be removed completely since its handled by readline already

**Comment by eatonphil** [comment]
> name 'printf' is not defined on macos thats weird.

Is this still an issue?

**Comment by Ghibranalj** [comment]
No it passed now. Idk what happened there or what fixed it.

**Comment by eatonphil** [comment]
If I type in nothing it shouldn't exit it should just go back to the beginning of the loop.

**Comment by eatonphil** [comment]
Here's an example of what I mean in psql. I hit enter four times and it just gave me another prompt

image

File: main.go [hunk 2]
```diff
@@ -219,6 +220,107 @@ func importFile(projectId string, file, mimetype string, ec runner.EvalContext)
...
+	filterInput := func(r rune) (rune, bool) {
+		switch r {
+		// block CtrlZ feature
+		case readline.CharCtrlZ:
+			return r, false
+		}
+		return r, true
+	}
+
+	defer os.Remove(tempfile.Name())
```
Comment (eatonphil) [inline main.go:279]: Just a nit, I normally prefer to keep cleanup code right after initialize code. I'd put this at line 259

**Comment by eatonphil** [comment]
Looks good, nice work!

**Pull request merged**